### PR TITLE
Reduce compile args to Im2Col kernel and add perf script

### PR DESCRIPTION
**Pull request by JehandadKhan**
Details: https://ontrack-internal.amd.com/browse/SWDEV-409657

[summary]:
Im2Col is re-compiled every time conv2d is called, 

`src/ocl/utilocl.cpp`:

These lines seem to be problematic:
```
        params += " -DNUM_CH_PER_WG=" + std::to_string(num_ch_per_wg);
        params += " -DNUM_IM_BLKS_X=" + std::to_string(num_blks_x);
        params += " -DNUM_IM_BLKS=" + std::to_string(num_blks);
        params += " -DLOCAL_MEM_SIZE=" + std::to_string(local_mem_sz);
        params += " -DSTRIDE_GT_1=" + std::to_string(static_cast<int>(stride_h * stride_w > 1));
        params += " -DTILE_SZ_X=" + std::to_string(tile_sz_x);
        params += " -DTILE_SZ_Y=" + std::to_string(tile_sz_y)
```

### CODE REVIEW

**Comment by JehandadKhan** [comment]
@junliume Can you please assign someone to write a GTest for this kernel?

**Comment by junliume** [comment]
> @junliume Can you please assign someone to write a GTest for this kernel?

@zjing14 could you assign Astha on this test supplementary part?

File: src/kernels/MIOpenIm2d2Col.cl [hunk 1]
```diff
@@ -124,9 +124,13 @@ kernel void Im2d2Col(const int data_size_off,
                      const int stride_w,
                      const int dilation_h,
                      const int dilation_w,
-                     global data_t* col)
+                     global data_t* col,
+                     const int num_ch_per_wg,
```
Comment (junliume) [inline src/kernels/MIOpenIm2d2Col.cl:128]: @JehandadKhan can `[[maybe_unused]]` be used in OCL codes?

```
[2023-08-10T05:13:35.160Z] /var/jenkins/workspace/MLLibs_MIOpen_jd_dyn_im2col/build/bin/test_immed_conv2d --bfloat16 --cmode conv --pmode default --group-count 1 --enable-fdb 0 --batch_size 16 --input_channels 32 --output_channels 32 --spatial_dim_elements 56 56 --filter_dims 3 3 --pads_strides_dilations 0 0 1 1 1 1 --trans_output_pads 0 0 --in_layout NCHW --fil_layout NCHW --out_layout NCHW 
[2023-08-10T05:13:35.160Z] FAILED: /var/jenkins/workspace/MLLibs_MIOpen_jd_dyn_im2col/src/hipoc/hipoc_program.cpp:304: Code object build failed. Source: MIOpenIm2d2Col.cl
[2023-08-10T05:13:35.160Z] Forward convolution: 
[2023-08-10T05:13:35.160Z] Input tensor: 16, 32, 56, 56
[2023-08-10T05:13:35.160Z] Weights tensor: 32, 32, 3, 3
[2023-08-10T05:13:35.160Z] Output tensor: 16, 32, 54, 54
[2023-08-10T05:13:35.160Z] Filter: conv2d, miopenConvolution, miopenPaddingDefault, {0, 0}, {1, 1}, {1, 1}, 
[2023-08-10T05:13:35.160Z] MIOpen(HIP): Error [Do] 'amd_comgr_do_action(kind, handle, in.GetHandle(), out.GetHandle())' AMD_COMGR_ACTION_COMPILE_SOURCE_WITH_DEVICE_LIBS_TO_BC: ERROR (1)
[2023-08-10T05:13:35.160Z] MIOpen(HIP): Error [BuildOcl] comgr status = ERROR (1)
[2023-08-10T05:13:35.160Z] MIOpen(HIP): Warning [BuildOcl] /tmp/comgr-646bda/input/MIOpenIm2d2Col.cl:128:32: error: unused parameter 'num_ch_per_wg' [-Werror,-Wunused-parameter]
[2023-08-10T05:13:35.160Z]                      const int num_ch_per_wg,
[2023-08-10T05:13:35.160Z]                                ^
[2023-08-10T05:13:35.160Z] 1 error generated.
```

**Review by junliume — APPROVED** [review]

File: src/kernels/MIOpenIm2d2Col.cl (hunk 1 above)
Comment (JehandadKhan) [reply src/kernels/MIOpenIm2d2Col.cl:128]: OpenCL is pure C and I am not sure if the `[[ ]]` attribute specification is valid there.

**Pull request merged**